app/controllers/comments_controller.rb

```diff
@@ -28,22 +30,27 @@ def create
   def update
     respond_to do |format|
       if @comment.update(comment_params)
+        @reference = Reference.find(@comment.root.commentable_id)
```

Comment:
If you're using `commentable_id`, you can just do `@reference = @comment.root.commentable`